app/payment-links/utils/payment-link-session.js

```diff
@@ -24,6 +24,10 @@ function setReference (req, productExternalId, reference, providedByQueryParams
   }
 }
 
+function removeReference (req, productExternalId) {
```

Comment:
As there are unit tests for this utility class, it would make sense to add a basic test for this.